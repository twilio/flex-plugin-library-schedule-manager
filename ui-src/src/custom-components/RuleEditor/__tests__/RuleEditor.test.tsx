import React from 'react';
import renderer from 'react-test-renderer';
import RuleEditor from '../RuleEditor';
const renderTree = (tree) => renderer.create(tree);
describe('<RuleEditor>', () => {
  it('should render component', () => {
    expect(
      renderTree(
        <RuleEditor
          onPanelClosed={() => {}}
          showPanel={false}
          schedules={[]}
          selectedRule={{
            id: 'eee7d8eb-f966-4a23-8743-ec3452cd11ad',
            name: 'July 4th',
            isOpen: false,
            closedReason: 'holiday',
            dateRRule: 'RRULE:FREQ=YEARLY;BYMONTH=7;BYMONTHDAY=4',
            startDate: '2023-01-24',
            endDate: '2023-01-24',
            startTime: '2023-01-24',
            endTime: '2023-01-24',
          }}
          onUpdateRule={() => {}}
        />,
      ).toJSON(),
    ).toMatchSnapshot();
  });
});
