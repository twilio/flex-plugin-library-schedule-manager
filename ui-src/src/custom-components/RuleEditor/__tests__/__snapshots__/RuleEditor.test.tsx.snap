// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RuleEditor> should render component 1`] = `null`;
